descartes_core/include/descartes_core/robot_model.h

```diff
@@ -129,7 +129,12 @@ class RobotModel
    * @return                 [description]
    */
   virtual bool isValidMove(const std::vector<double> &from_joint_pose, const std::vector<double> &to_joint_pose,
-                           double dt) const = 0;
+                           double dt) const
+  {
+    return isValidMove(from_joint_pose.data(), to_joint_pose.data(), dt);
+  }
+
+  virtual bool isValidMove(const double* s, const double* f, double dt) const = 0;
```

Comment:
Perhaps we could have the pointer based impl default to copying data into vectors and calling the original function. We could then specialize the pointer func in MoveitStateAdapter to just use the raw data.

The vector copies are expensive though, so if this new planning_graph went into the mainstream and you weren't using MoveitStateAdapter, you're code would probably get several times slower.